### DISCUSSION TITLE
Asset construction with payloads.

### DISCUSSION THREAD

**Discussion by sanders3d** [Autodesk/maya-usd#3446] [Q&A]
So I'm doing my best to try and create assets in Maya as we are used to do them in Houdini. This means we follow closely this guide: https://github.com/usd-wg/assets/blob/main/docs/asset-structure-guidelines.md

I've gotten to the part where I'm able to create the asset container which has a default primitive and also a class object. I've also added a payload. But there is where it stops ( all done in python ). Perhaps I'm doing this the wrong way, but what I'm trying to create is something that looks like this structure
![image](https://github.com/Autodesk/maya-usd/assets/44543707/f010df12-2b24-44b5-9378-f7b17eb33d84)

asset.usda 
contains the class object and the link to the payload. It should also contain the bounding box once I have the geometry.

payload.usda
this holds the sublayers for geometry and shading ( geom.usdc, shade.usda )

Via the layer editor I'm not really able to compose the asset in this fashion. Maybe its a limitation but I'm not sure how else I would go about it.

Which brings up the next question. Should we disregard the layer editor completely for asset construction? Would this be where we only use bifrost? ( which I have yet to explore ).

regards
stefan

**Comment by sharktacos** [comment]
FWIW, I took a similar approach here: https://github.com/Autodesk/maya-usd/discussions/3340

The difference I believe is that I am not loading assets/components in the layer editor, but rather am using that for all the work from each department (modeling, layout, lookdev, each has their own sublayer). Then  I use the Outliner to build the stage, referencing or payloading in components/assets. 

I have a video walkthrough of the process here: https://www.youtube.com/watch?v=mIGvq5-7HAI&t=809s

Hope it's helpful,
Derek
